inst/@sym/horzcat.m

```diff
@@ -62,7 +62,7 @@
           'return sp.Matrix.hstack(*_proc),'
           };
 
-  varargin = sym(varargin);
+  varargin = cell_array_to_sym(varargin);
```

Comment:
IOW, I think the current situation of

    my_syms = sym ({'x', 'y', 'z'})

is neither good form nor does it seem to be legal under `classdef`-defined classes. A call to `sym (anything)` should always return an object of type `sym`.

Using `classdef`, it might be better to define a static method such as

    my_syms = sym.symarray ({'x', 'y', 'z'})